### DISCUSSION TITLE
Should Namespace and Cluster default views be some kind of summary dashboards

### DISCUSSION THREAD

**Discussion by mklanjsek** [vmware-tanzu/octant#1745] [Ideas]
I presented this idea on today's standup, will try to summarize it here to flesh it out and collect more feedback.

Basically I feel that we may need a better default pages for both `Namespace` and `Cluster` modules. This became obvious after the latest navigation change. First problem is that switching back and forth between views in these two modules takes a lot of time since we show all details inside the tables of default landing pages and that can be a lot of information for large clusters.  Additionally I am questioning if that should be the landing page we would use for these two modules - displaying a dashboard that gives summary visual representation that's clear at a glance would be IMHO better user experience.  Maybe something like this:

![Screen Shot 2020-12-16 at 11 35 28 AM](https://user-images.githubusercontent.com/34245594/102532089-a8ebdd00-4060-11eb-905e-79b3b89d724b.png)
  
Instead of Donut charts, we can use a New Resource Viewer view displaying similar info. 

The current `Namespace Overview` and `Cluster Overview` pages  should still be available but can be renamed to something like `Namespace Details` and `Cluster Details`. 

I feel this would improve the user experience and in same time improve the loading time of default views.

**Comment by mklanjsek** [comment]
Here is more ad-hoc discussion about this:

wwitzel3  11:42 AM
so I was thinking about the namespace overview, the intention of it is to display everything in a namespace
I think we might be trying to solve one problem in two different ways
and maybe we need to make Applications our default view and fix that view to be better
Workloads doesn't include things like config maps and secrets etc.. and the Namespace Overview does

Milan Klanjsek  11:46 AM
That makes sense, but question still is is that the best view to display when you click on Namespace Overview

wwitzel3  11:46 AM
I think yes, here is why, I just stepped through a very common thing I do, which is pull up the namespace view and then click a label so I can see everything with that label
config maps, secrets, pods, deployments, etc.. it lets me get a view of "everything" and if we change the overview, there is no place to get that anymore

Milan Klanjsek  11:49 AM
Then what if we additional view called Namespace Details containing everything that Namespace Overview  has today?

wwitzel3  11:50 AM
I think that's just trying to solve the same problem twice, if applications becomes the default view and loads up the resources and donut charts, we can leave namespace overview alone, I think we can rename "Overview" to "Details"

Milan Klanjsek  11:53 AM
Sorry, don’t agree I see Applications as applications dashboard, but I still feel that we need similar kind of dashboard for Namespaces and Cluster.

wwitzel3  11:55 AM
Octant is a developer tool, as a developer I care about my application, so that should be the default view. If something is wrong with my application, I want some way to drill down in to the details.
While I think the view you made looks nicer, as a developer it is less useful.
As an ops view, I think it is very useful

sfoo:vmware:  11:56 AM
can we move this to the public slack?

wwitzel3  11:56 AM
Yes.

Milan Klanjsek  11:57 AM
Fair point. Just wanted to point out that the Applications view will not show for example total number of pods in specific namespace and total number of Nodes in a cluster. Also number of errors in each…

I’ll open a discussion item in git and copy all this there

wwitzel3  11:58 AM
:thumbsup:
Thank you Milan

**Comment by mklanjsek** [comment]
Here is also a [link](https://gist.github.com/mklanjsek/3b01caa548e8b666f1382d4da334a822) to JSON used for the above prototype, in case someone wants to play with it in Octant Sandbox.

**Comment by mklanjsek** [comment]
@vmware-tanzu/octant-owners any thoughts?

**Comment by ibagha** [comment]
I do agree with Milan that there could be more value added to the namespace and cluster overviews if we didn't leave the user to parse through the tables which could get dense (not to mention there's content concealed in each table). A couple things here:

1. There are two dashboard-type visuals now in Octant: the doughnut view used in applications and the resource viewer. We should be intentional and have loose guidelines for where we use what view. What do you all see as the use cases for using the doughnut view vs resource viewer?
2. It'd be helpful to capture what is the user pain by not having a visual summary up top in the namespace and cluster overviews, as well as quantify how many users have requested this change. Are there and Slack threads or open issues that we can link here to capture this?

**Comment by mklanjsek** [comment]
Thank you @ibagha! The idea here is to give summary view that's easy to read at a glance and renders fast, so if new resource viewer achieves that, it would be my preferred solution. The question we need to ask is:

What is the info user cares most about namespace/cluster?  

If answer is number and state of all resources then donut chart would work best, if topology is more important, that it would be the resource viewer.  I am hoping that the new resource viewer would eventually be able to show all that info in visually more pleasing way.

I feel we should test this idea initially with donut charts to get feedback about usability and if that's positive, switch to RV once it's available.
